packages/orbit-components/src/tmp-tailwind/ButtonLink/index.tsx

```diff
@@ -0,0 +1,107 @@
+};
+
+const typeCompactStyles: Record<Type, string> = {
+  primary:
+    "text-button-link-primary-foreground focus:text-button-link-primary-foreground-focus active:text-button-link-primary-foreground-active hover:text-button-link-primary-foreground-hover",
+  secondary:
+    "text-button-link-secondary-foreground focus:text-button-link-secondary-compact-foreground-hover active:text-button-link-secondary-compact-foreground-active hover:text-button-link-secondary-compact-foreground-hover",
+  critical:
+    "text-button-link-critical-foreground focus:text-button-link-critical-foreground-focus active:text-button-link-critical-foreground-active hover:text-button-link-critical-foreground-hover",
+};
+
+const typeDisabledStyles: Record<Type, string> = {
+  primary: "bg-button-link-primary-background text-button-link-primary-foreground",
+  secondary: "bg-button-link-secondary-background text-button-link-secondary-foreground",
+  critical: "bg-button-link-critical-background text-button-link-critical-foreground",
+};
+
+const typeCompactDisabledStyles: Record<Type, string> = {
+  primary: "text-button-link-primary-foreground",
+  secondary: "text-button-link-secondary-foreground",
+  critical: "text-button-link-critical-foreground",
+};
+
+const ButtonLink = React.forwardRef<HTMLButtonElement, Props>(
+  (
+    {
+      type = "primary",
+      size = "normal",
+      compact,
+      disabled,
+      children,
+      iconLeft,
+      iconRight,
+      ...props
+    },
+    ref,
+  ) => {
+    return (
+      <ButtonPrimitive
+        ref={ref}
+        iconLeft={iconLeft}
+        iconRight={iconRight}
+        {...props}
+        className={cx(
+          "orbit-button space-x-xs rtl:space-x-reverse",
+          sizeStyles[size],
+          compact !== true && disabled !== true && typeStyles[type],
+          compact === true && disabled !== true && typeCompactStyles[type],
+          compact !== true && disabled === true && typeDisabledStyles[type],
+          compact === true && disabled === true && typeCompactDisabledStyles[type],
+          compact !== true &&
+            children != null &&
+            iconLeft == null &&
+            iconRight == null &&
+            paddingNoIconsStyles[size],
+          compact !== true &&
+            children != null &&
+            iconLeft != null &&
+            iconRight == null &&
+            paddingLeftIconStyles[size],
+          compact !== true &&
+            children != null &&
+            iconLeft == null &&
+            iconRight != null &&
+            paddingRightIconStyles[size],
+          compact !== true &&
+            children != null &&
+            iconLeft != null &&
+            iconRight != null &&
+            paddingBothIconsStyles[size],
```

Comment:
This is not correct. And I have just noticed we have the same problem on Button 😞 As an example, a large button with icon on both sides should not have a 24px horizontal padding, but a 16px.

Reply:
adjusted, nice catch, yea it was difficult to get these right, and are not immedatelly obvious which ones are correct and which are not